controllers/rhmi/rhmi_controller.go

```diff
@@ -1419,3 +1441,63 @@ func isInstallationOlderThan1Minute(installation *rhmiv1alpha1.RHMI) bool {
 	aMinuteAgo := time.Now().Add(-time.Minute)
 	return aMinuteAgo.After(installation.CreationTimestamp.Time)
 }
+
+func validateAddOnStsRoleArnParamaterPattern(client k8sclient.Client, namespace string) (bool, error) {
+	// function is checking if STS addon parameter Pattern is valid
+	// Parameter is Valid only in case:
+	// 1.	Parameter exists and value matching AWS Role ARN pattern
+	// Parameter is Not valid  in other cases:
+	// 2.	parameter exists and value is NOT matching AWS Role ARN pattern
+	// 3.	parameter exists and value is empty
+	// 4.	parameter does not exists
+
+	awsArnPattern := "arn:aws:iam:\\S*:\\d+:role\\/\\S+"
```

Comment:
We can future-proof this regexp pattern if we take other AWS partitions into account.
Particularly US GovCloud - https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-arns.html
```suggestion
	awsArnPattern := "arn:aws(?:-us-gov)?:iam:\\S*:\\d+:role\\/\\S+"
```

Reply:
Done. Thank you
Just syntax is not so clear for me (probably I need look more at regular expressions),
for me something like following is more clear/readable : )        (aws)|(aws-us-gov)
But your syntax is working fine , I checked, so using it : )

Reply:
You are right, this way - `(aws)|(aws-us-gov)`
it is probably more readable. Feel free to change it.